benchmarks/spinlock_bench/spinlock_bench.c

```diff
@@ -0,0 +1,107 @@
+/****************************************************************************
+ * apps/benchmarks/spinlock_bench/spinlock_bench.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <nuttx/spinlock.h>
+#include <time.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define TEST_NUM CONFIG_SPINLOCK_MULTITHREAD
+#define THREAD_NUM CONFIG_SPINLOCK_MULTITHREAD
+
+/****************************************************************************
+ * Private Type
+ ****************************************************************************/
+
+struct thread_parmeter_s
+{
+  FAR int *result;
+  FAR spinlock_t *lock;
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static FAR void *thread_spinlock(FAR void *parameter)
+{
+  FAR int *result = ((struct thread_parmeter_s *)parameter)->result;
+  FAR spinlock_t *lock = ((struct thread_parmeter_s *)parameter)->lock;
+
+  for (int i = 0; i < TEST_NUM; i++)
+    {
+      spin_lock(lock);
+      (*result)++;
+      spin_unlock(lock);
+    }
+
+  return NULL;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void main(void)
+{
+  spinlock_t lock = SP_UNLOCKED;
+  int result = 0;
+  pthread_t thread[THREAD_NUM];
+  struct thread_parmeter_s para;
+  clock_t start, end;
```

Comment:
split to.two line

Reply:
Done. Please review.
Thank you a lot!!!